mapper/tedge_mapper/src/main.rs

```diff
@@ -1,14 +1,35 @@
+use crate::error::MapperError;
+use crate::time_provider::SystemTimeProvider;
+use flockfile::{Flockfile, FlockfileError};
 use mqtt_client::Client;
+use std::path::PathBuf;
+use tedge_config::{
+    AzureMapperTimestamp, ConfigRepository, ConfigSettingAccessor, TEdgeConfigRepository,
+};
 use tracing::{debug_span, error, info, Instrument};
 
+mod az_mapper;
+mod c8y_mapper;
+mod error;
 mod mapper;
+mod time_provider;
 
-const APP_NAME: &str = "tedge-mapper";
-const DEFAULT_LOG_LEVEL: &str = "warn";
+const APP_NAME_C8Y: &str = "tedge-mapper-c8y";
+const APP_NAME_AZ: &str = "tedge-mapper-az";
+const DEFAULT_LOG_LEVEL: &str = "debug";
 const TIME_FORMAT: &str = "%Y-%m-%dT%H:%M:%S%.3f%:z";
 
 #[tokio::main]
-async fn main() -> Result<(), anyhow::Error> {
+async fn main() -> Result<(), MapperError> {
+    let args: Vec<String> = std::env::args().collect();
+
+    // Only one argument is allowed
+    if args.len() != 2 {
+        return Err(MapperError::IncorrectArgument);
+    }
+
+    let cloud_name = args[1].as_str();
```

Comment:
You could add a `enum CloudName { C8y, Azure }` and a `FromStr` impl. Then you write:

```let cloud_name = CloudName::from_str(args[1].as_str())?```